### PR TITLE
15 dark colorscheme

### LINKED ISSUE #15: Add dark colorscheme
It would be a lot more pleasing for some users, to have a dark theme and an
option to switch between themes.

### PR DESCRIPTION
**Pull request by GlancingMind**
Might close #15 when merged.
Most color values use only the text-palette as this is the palette which is adjustable via the palette.type flag as described [here](https://material-ui.com/customization/palette/#dark-mode).
This means that changes to the "colorful"-palettes won't be reflected on the overall theme.

Currently it looks like, that a generic "light/dark"-Mode is incompatible with the colorscheme-editor. Unless the user only focuses on the palette.text, palette.divider and palette.background values. Which will prevent the text to be grayish.
Maybe this current implementation has to be dropped for multiple separate themes and the mode-switch should only refer to the two hard-coded themes. The problem with this might be, that the whole color-palette must be alterable by the color-editor, as the user doesn't know which specific palette-field is used. E.g. deeporange['300']. May be only the gray palette has to be changed...

TLDR - Two possible solutions:
- Implement colorscheme-editor with only field for the above mentioned light/dark convertible color palettes.
- Implement light and dark mode via two separate themes and hard-code them for the mode-switch. User defined dark or light themes are not usable from the mode-switch only the default ones. Might also have to specify the whole gray-palette...

### CODE REVIEW

**Comment by MichaelMure** [comment]
Hey @GlancingMind, git-bug's author here. I noticed your progress on this fork. Have you considered contributing those changes directly to the git-bug repo? The webUI changes you are working on and planned are definitely interesting and it it would be great to have them available for everyone.

**Comment by GlancingMind** [comment]
@MichaelMure Oh hi :-)
Well, that was the plan if everything goes well.
But currently this is a university course project and our deadline is set for the end of march.
I think that our tutors are fine with contributing this changes before the deadline, but I rather like to ask them again.
And of course if my fellow students agree for their respective changes. I already have their permission, but I rather like to ask them again, too. Just to be sure.

**Comment by MichaelMure** [comment]
That's cool! I'd suggest to try to upstream your changes once they are ready rather than at the end of the project if you can, to avoid conflicts if the two repository diverge too much. Also, it's better for me to deal with multiple small PR than a huge dunk at the end. This might also avoid duplicate effort between your team and @claudioantonio especially.

**Comment by LenaNoctua** [comment]
If I manage to get anything done, I would agree, too.

**Comment by claudioantonio** [comment]
Hi! This issue is related to https://github.com/MichaelMure/git-bug/issues/573. ;-)

**Comment by GlancingMind** [comment]
@MichaelMure Hi Michael I got the permission to upstream our changes before the deadline :tada:.
I already created a PR for our container setup [here](https://github.com/MichaelMure/git-bug/pull/576).

@claudioantonio Nice work and thank you for the hints. Though as this is a university project for a course, I am not sure, if we are allowed to use this current upstream implementation of yours. It might happen, that we have to implement this Issues our-self, too. So possible divergence ahead, but this shouldn't concern you or Michael.

**Comment by GlancingMind** [comment]
This should work. Light theme hasn't changed significantly.

image
image

**Comment by GlancingMind** [comment]
Close in favor of #27
